include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_radix_sort_one_wg.h

```diff
@@ -0,0 +1,176 @@
+// -*- C++ -*-
+//===----------------------------------------------------------------------===//
+//
+// Copyright (C) Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// This file incorporates work covered by the following copyright and permission
+// notice:
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef _ONEDPL_parallel_backend_sycl_radix_sort_one_wg_H
+#define _ONEDPL_parallel_backend_sycl_radix_sort_one_wg_H
+
+//namespace oneapi
+//{
+//namespace dpl
+//{
+//namespace __par_backend_hetero
+//{
+
+template <int ITEMS_PER_THREAD, typename KeyT, typename _Wi, typename _Src, typename _Keys>
+void
+__block_load(const _Wi __wi, const _Src& __src, _Keys& __keys, const uint32_t __n)
+{
+    constexpr KeyT __default_key = (KeyT)0xffffffff;
+
+    #pragma unroll
+    for (auto i = 0; i < ITEMS_PER_THREAD; i++)
+    {
+        const uint32_t __offset = __wi*ITEMS_PER_THREAD + i;
+        //boundary check is slow but nessecary
+        if (__offset < __n)
+            __keys[i] = __src[__offset];
+        else
+            __keys[i] = __default_key;
+    }
+}
+
+template <int ITEMS_PER_THREAD, typename _Item, typename _Wi, typename _Lacc, typename _Keys, typename _Ranks>
+void
+__to_blocked(_Item __it, const _Wi __wi, _Lacc& __exchange_lacc, _Keys& __keys, const _Ranks& __ranks)
+{
+    for (int i = 0; i < ITEMS_PER_THREAD; i++)
+        __exchange_lacc[__ranks[i]] = __keys[i];
+
+    __dpl_sycl::__group_barrier(__it);
+
+    for (int i = 0; i<ITEMS_PER_THREAD; i++)
+        __keys[i] = __exchange_lacc[__wi*ITEMS_PER_THREAD + i];
```

Comment:
This loop logic might result in bank conflicts. For example, we may be limited to 1/16 of potential local memory bandwidth for 32bit keys and `ITEMS_PER_THREAD`=16 if bank conflicts occur with read operations.
I have no ready solution if so. Read from`__exchange_lacc` with `ITEMS_PER_THREAD` stride and then exchange elements in `keys` between work-items in the same sub-groups? This is too difficult.

Reply:
Probably upcoming "load, store" SYCL API will solve the problem.

Reply:
In any case I suggest solving that potential problem in one of the next code update, not now.